dvc/repo/experiments/__init__.py

```diff
@@ -162,13 +162,25 @@ def _config_clone(self):
     def _scm_checkout(self, rev):
         self.scm.repo.git.reset(hard=True)
         if self.scm.repo.head.is_detached:
-            # switch back to default branch
-            self.scm.repo.heads[0].checkout()
```

Comment:
`heads` ends up being sorted alphabetically by gitpython, rather than returning branches in the order they were generated. So if `heads[0]` was an experiments-workspace-only branch, the subsequent `git pull` would end up failing, since there was associated no upstream tracking branch.